runtime/lua/vim/diagnostic.lua

```diff
@@ -1824,8 +1824,13 @@ function M.open_float(opts, ...)
     -- does not have a dedicated table for configuration options; instead, the options are mixed in
     -- with its `opts` table which also includes "keyword" parameters. So we create a dedicated
     -- options table that inherits missing keys from the global configuration before resolving.
-    local t = global_diagnostic_options.float
```

Comment:
can't we just change this to:

    local t = global_diagnostic_options.float or global_diagnostic_options.float(0, bufnr)

Reply:
yes i forget it use `if_nil` to set a default value always.